src/preferences/colorpaletteeditormodel.h

```diff
@@ -46,3 +49,27 @@ class ColorPaletteEditorModel : public QStandardItemModel {
     bool m_bEmpty;
     bool m_bDirty;
 };
+
+class HotcueIndexListItem : public QStandardItem {
+    Q_OBJECT
+  public:
+    const QRegularExpression hotcueListMatchingRegex(QStringLiteral("([0-9]+)[ ,]+"));
+
+    HotcueIndexListItem(const QList<int>& hotcueList = {});
+
+    void setData(const QVariant& value, int role = Qt::UserRole + 1) override;
```

Comment:
I still don't quite understand the purpose of roles. I get that `setData` and `data` are "generic" accessors to get data for different purposes (such as displaying the data itself, providing a tooltip description etc). Since I only intent to intercept the data for editing and displaying, I should be able to reuse those roles instead of making my own role, right? Or what exactly am I supposed to do with the custom role.

Reply:
Right.